normalize_label.go

```diff
@@ -38,10 +39,22 @@ type LabelNamer struct {
 //
 // If UTF8Allowed is true, the label is returned as is. This option is provided just to
 // keep a consistent interface with the MetricNamer.
-func (ln *LabelNamer) Build(label string) string {
+func (ln *LabelNamer) Build(label string) (normalizedName string, err error) {
+	defer func() {
+		// Check that the resulting normalized name contains at least one non-underscore character
+		for _, c := range normalizedName {
+			if c != '_' {
+				return
+			}
+		}
+		normalizedName = ""
+		err = errors.New("label name normalization resulted in invalid name")
+	}()
+
 	// Trivial case.
 	if len(label) == 0 || ln.UTF8Allowed {
```

Comment:
this will now return an error if the input was blank. not sure if we want to do that

Reply:
Are empty label keys valid in prometheus? I didn't think they were

Reply:
I am guessing a blank label was getting caught elsewhere, so this should only result in a better reason for why the error happened.